src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -1569,51 +1549,66 @@ void GetVRFInputData(EnergyPlusData &state, bool &ErrorsFound)
-        state.dataHVACVarRefFlow->TerminalUnitList(VRFNum).HeatingCoilPresent = true;
-        state.dataHVACVarRefFlow->TerminalUnitList(VRFNum).TerminalUnitNotSizedYet = true;
-        state.dataHVACVarRefFlow->TerminalUnitList(VRFNum).HRHeatRequest = false;
-        state.dataHVACVarRefFlow->TerminalUnitList(VRFNum).HRCoolRequest = false;
-        state.dataHVACVarRefFlow->TerminalUnitList(VRFNum).CoolingCoilAvailable = false;
-        state.dataHVACVarRefFlow->TerminalUnitList(VRFNum).HeatingCoilAvailable = false;
-        state.dataHVACVarRefFlow->TerminalUnitList(VRFNum).CoolingCoilAvailSchPtr = -1;
-        state.dataHVACVarRefFlow->TerminalUnitList(VRFNum).HeatingCoilAvailSchPtr = -1;
-
-        for (TUListNum = 1; TUListNum <= state.dataHVACVarRefFlow->TerminalUnitList(VRFNum).NumTUInList; ++TUListNum) {
-            state.dataHVACVarRefFlow->TerminalUnitList(VRFNum).ZoneTUName(TUListNum) = cAlphaArgs(TUListNum + 1);
-        }
-    }
+        auto &thisTUList = state.dataHVACVarRefFlow->TerminalUnitList(TUListNum);
+        thisTUList.Name = cAlphaArgs(1);
+        thisTUList.NumTUInList = NumAlphas - 1;
+        thisTUList.ZoneTUPtr.allocate(thisTUList.NumTUInList);
+        thisTUList.ZoneTUName.allocate(thisTUList.NumTUInList);
+        thisTUList.IsSimulated.allocate(thisTUList.NumTUInList);
+        thisTUList.TotalCoolLoad.allocate(thisTUList.NumTUInList);
+        thisTUList.TotalHeatLoad.allocate(thisTUList.NumTUInList);
+        thisTUList.CoolingCoilPresent.allocate(thisTUList.NumTUInList);
+        thisTUList.HeatingCoilPresent.allocate(thisTUList.NumTUInList);
+        thisTUList.TerminalUnitNotSizedYet.allocate(thisTUList.NumTUInList);
+        thisTUList.HRHeatRequest.allocate(thisTUList.NumTUInList);
+        thisTUList.HRCoolRequest.allocate(thisTUList.NumTUInList);
+        thisTUList.CoolingCoilAvailable.allocate(thisTUList.NumTUInList);
+        thisTUList.HeatingCoilAvailable.allocate(thisTUList.NumTUInList);
+        thisTUList.CoolingCoilAvailSchPtr.allocate(thisTUList.NumTUInList);
+        thisTUList.HeatingCoilAvailSchPtr.allocate(thisTUList.NumTUInList);
+        thisTUList.ZoneTUPtr = 0;
+        thisTUList.IsSimulated = false;
+        thisTUList.TotalCoolLoad = 0.0;
+        thisTUList.TotalHeatLoad = 0.0;
+        thisTUList.CoolingCoilPresent = true;
+        thisTUList.HeatingCoilPresent = true;
+        thisTUList.TerminalUnitNotSizedYet = true;
+        thisTUList.HRHeatRequest = false;
+        thisTUList.HRCoolRequest = false;
+        thisTUList.CoolingCoilAvailable = false;
+        thisTUList.HeatingCoilAvailable = false;
+        thisTUList.CoolingCoilAvailSchPtr = -1;
+        thisTUList.HeatingCoilAvailSchPtr = -1;
+
+        for (int TUNum = 1; TUNum <= thisTUList.NumTUInList; ++TUNum) {
+            thisTUList.ZoneTUName(TUNum) = cAlphaArgs(TUNum + 1);
+        }
+    }
+
+    auto checkCurveMinMaxOutput = [&state](int CurveIndex) -> std::array<Real64, 4> {
```

Comment:
Man, you are really loving this lambda trick.

Reply:
It avoids duplicating the code, yet making a full function is too much, so a lambda is a good candidate no?

Reply:
I'm just saying.